pkg/prom/diagnostics.go

```diff
@@ -0,0 +1,68 @@
+package prom
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/kubecost/cost-model/pkg/env"
+	"github.com/kubecost/cost-model/pkg/log"
+	prometheus "github.com/prometheus/client_golang/api"
+)
+
+// QueuedPromRequest is a representation of a request waiting to be sent by the prometheus
+// client.
+type QueuedPromRequest struct {
+	Context   string `json:"context"`
+	Query     string `json:"query"`
+	QueueTime int64  `json:"queueTime"`
```

Comment:
I think adding a unit (seconds, nanoseconds, ...) to the name of this field would be very helpful.